src/topology/algebra/module/star.lean

```diff
@@ -22,6 +22,19 @@ def starL (R : Type*) {A : Type*}
   continuous_to_fun := continuous_star,
   continuous_inv_fun := continuous_star }
 
+/-- If `A` is a topological module over a commutative `R` with trivial star and compatible actions,
+then `star` is a continuous linear equivalence. -/
+@[simps]
+def starL' (R : Type*) {A : Type*}
+  [comm_semiring R] [star_ring R] [has_trivial_star R] [add_comm_monoid A] [star_add_monoid A]
+  [module R A] [star_module R A] [topological_space A] [has_continuous_star A] :
+    A ≃L[R] A :=
+(starL R : A ≃L⋆[R] A).trans
+  ({ map_smul' := λ r a, by simp [star_ring_end_apply],
+    continuous_to_fun := continuous_id,
+    continuous_inv_fun := continuous_id,
+    ..add_equiv.refl A, } : A ≃L⋆[R] A)
```

Comment:
[Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Star-semilinear.20maps.20are.20semilinear.20when.20star.20is.20trivial/near/359552581)

Reply:
If you add
```
instance has_trivial_star.ring_hom_comp_triple_left (R : Type u) [comm_semiring R]
  [star_ring R] [has_trivial_star R] :
  ring_hom_comp_triple (star_ring_end R) (ring_hom.id R) (ring_hom.id R) :=
⟨by ext; exact star_trivial _⟩

instance has_trivial_star.ring_hom_comp_triple_right (R : Type u) [comm_semiring R]
  [star_ring R] [has_trivial_star R] :
  ring_hom_comp_triple (ring_hom.id R) (star_ring_end R) (ring_hom.id R) :=
⟨by ext; exact star_trivial _⟩
```

You can do 
```
@[simps]
def starL' (R : Type*) {A : Type*}
  [comm_semiring R] [star_ring R] [has_trivial_star R] [add_comm_monoid A] [star_add_monoid A]
  [module R A] [star_module R A] [topological_space A] [has_continuous_star A] :
    A ≃L[R] A :=
(starL R : A ≃L⋆[R] A).trans (1 : A ≃L[R] A)
```

(and at that point maybe it's better to use `(starL R : A ≃L⋆[R] A).trans (1 : A ≃L[R] A)` directly and not define `starL'`)

Reply:
I liked the idea of reusing the current `ring_hom_comp_triple` infrastructure because it would have been nice when composing a lot of things if restriction of scalars was done automatically. So I tried to express your construction in these terms, and I am now convinced that it isn't possible for two reasons.

First, we would need `algebra R₁' R₁` so that your hypothesis `h` really becomes a commutative diagram of ring homs. I think this wouldn't be a big problem in practice, but still it's a bit annoying.

The second much more decisive issue is that composition easily turns `(f : M →ₛₗ[σ] N)` into `(f : M →ₛₗ[σ.comp (algebra_map R₁' R₁)] N)`, but then you can't get to `M →ₛₗ[σ'] N` because you would need to postcompose by the identity of `N` as a `inv_fun (algebra_map R₁' R₁)`-linear map, but `(algebra_map R₁' R₁)` need not be invertible so you can't express that (hope what I say makes sense).

Anyway, I now agree that your solution is the right one.